webapp/src/pages/eventConfig/components/EventConfigComponent.js

```diff
@@ -271,12 +289,102 @@ class EventConfigComponent extends Component {
         isValid: errors.length === 0
       });
     });
+    console.log(this.state.updatedEvent);
   }
 
   getFieldNameWithLanguage = (input, lang) => {
     return input + " in " + lang;
   }
 
+  setRequiredDateFields = (event_type) => {
+    //could shift this const globally?
+    const requiredDateFields_by_event = {
+      "Event": ALL_DATE_FIELDS,
+      "Programme": ALL_DATE_FIELDS,
+      "Award": ALL_DATE_FIELDS.slice(0, 4),
+      "Call": ALL_DATE_FIELDS.slice(0, 3)
+    }
+    const requiredDateFields = requiredDateFields_by_event[event_type];
+    const future_date = new Date();
+    future_date.setFullYear(2099);
+    this.setState({
+      requiredDateFields: requiredDateFields,
+      //sets all unrequired dates to a future date, and all required dates to ''
+      ...ALL_DATE_FIELDS.forEach(date => {
+        const updated_date = !requiredDateFields.includes(date) ? this.formatDate(future_date) : ""; //ensures all dates are reset each time user changes dropdown menu
+          this.setState(
+            {...this.state.updatedEvent,
+              [date]: updated_date
+            }
+          )
+      })
```

Comment:
This is not working. I tried many variations, but the date fields are just not updated - any ideas?

Reply:
Alright, I finally got this working (see latest commit) though not sure if this is the best place for the call - let me know if you'd recommend another way :)